src/Compilers/CSharp/Portable/Binder/BlockBinder.cs

```diff
@@ -56,5 +56,22 @@ internal override ImmutableArray<LocalSymbol> GetDeclaredLocalsForScope(CSharpSy
 
             throw ExceptionUtilities.Unreachable;
         }
+
+        internal override ImmutableArray<LocalFunctionSymbol> GetDeclaredLocalFunctionsForScope(CSharpSyntaxNode node)
+        {
+            if (node.Kind() == SyntaxKind.Block)
+            {
+                if (((BlockSyntax)node).Statements == _statements)
+                {
+                    return this.LocalFunctions;
+                }
+            }
+            else if (_statements.Count == 1 && _statements.First() == node)
+            {
+                return this.LocalFunctions;
+            }
+
+            throw ExceptionUtilities.Unreachable;
```

Comment:
I assume it's because that if we don't find the syntax in the current block, then something has gone horribly wrong (the parameter is invalid - we should never pass in a syntax node that isn't the exact block the binder is located at).